### PR TITLE
workaround: disable tests on github actions with windows runner

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

**Review by itsacoderepo — APPROVED** [review]

File: console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala [hunk 1]
```diff
@@ -40,91 +40,62 @@ class CPGQLServerTests extends AnyWordSpec with Matchers {
...
+    * io.undertow.server.HttpServerExchange endExchange ERROR: UT005090: Unexpected failure
+    * java.lang.NoClassDefFoundError: Could not initialize class org.xnio.channels.Channels at
+    * io.undertow.io.UndertowOutputStream.close(UndertowOutputStream.java:348)
+    *
+    * This happens for both windows 2019 and 2022, and isn't reproducable elsewhere. Explicitly adding a dependency on
+    * `org.jboss.xnio/xnio-api` didn't help, as well as other debug attempts. So we gave up and disabled this
+    * specifically for github actions' windows runner.
+    */
+  val isGithubActions = System.getenv("GITHUB_ACTIONS") == "TRUE"
+  val isWindows       = System.getProperties.getProperty("os.name", "<unknown>").toLowerCase.contains("windows")
```
Comment (max-leuthaeuser) [inline console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala:55]: You may use `scala.util.Properties.isWin` here.

File: console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala [hunk 2]
```diff
@@ -40,91 +40,62 @@ class CPGQLServerTests extends AnyWordSpec with Matchers {
...
+    * 3:08:54 PM org.jboss.threads.Version <clinit> INFO: JBoss Threads version 3.1.0.Final Apr 21, 2022 3:08:55 PM
+    * io.undertow.server.HttpServerExchange endExchange ERROR: UT005090: Unexpected failure
+    * java.lang.NoClassDefFoundError: Could not initialize class org.xnio.channels.Channels at
+    * io.undertow.io.UndertowOutputStream.close(UndertowOutputStream.java:348)
+    *
+    * This happens for both windows 2019 and 2022, and isn't reproducable elsewhere. Explicitly adding a dependency on
+    * `org.jboss.xnio/xnio-api` didn't help, as well as other debug attempts. So we gave up and disabled this
+    * specifically for github actions' windows runner.
+    */
+  val isGithubActions = System.getenv("GITHUB_ACTIONS") == "TRUE"
```
Comment (max-leuthaeuser) [inline console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala:54]: Safer variant: `scala.util.Properties.envOrElse`

**Review by max-leuthaeuser — APPROVED** [review]
Small nits.

File: console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala (hunk 1 above)
Comment (mpollmeier) [reply console/src/test/scala/io/joern/console/cpgqlserver/CPGQLServerTests.scala:55]: nice, didn't know this one, thanks for sharing

**Comment by max-leuthaeuser** [comment]
Also we can re-use scalatest tags for selectively disabling tests like I did here:
https://github.com/joernio/joern/blob/master/console/src/test/scala/io/joern/console/PluginManagerTests.scala#L16

**Pull request merged**